packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -99,6 +102,38 @@ export interface IDeltaManagerInternalEvents extends IDeltaManagerEvents {
     (event: "closed", listener: (error?: ICriticalContainerError) => void);
 }
 
+/**
+ * Implementation of IDocumentDeltaConnection that does not support submitting
+ * or receiving ops. Used in storage-only mode.
+ */
+class NoDeltaStream extends TypedEventEmitter<IDocumentDeltaConnectionEvents> implements IDocumentDeltaConnection {
+    clientId: string = undefined as any;
```

Comment:
There is assumption somewhere in the code that clientId is not empty string. I think some test will fail. Even if not, it's better to adhere to this invariant. I've run it with this PR: https://github.com/microsoft/FluidFramework/pull/5104/files (see changes in drivers).